stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser2/adapters/EventAdapter.kt

```diff
@@ -0,0 +1,226 @@
+import io.getstream.chat.android.client.api2.model.dto.MessageUpdatedEventDto
+import io.getstream.chat.android.client.api2.model.dto.NewMessageEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationAddedToChannelEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationChannelDeletedEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationChannelMutesUpdatedEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationChannelTruncatedEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationInviteAcceptedEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationInvitedEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationMarkReadEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationMessageNewEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationMutesUpdatedEventDto
+import io.getstream.chat.android.client.api2.model.dto.NotificationRemovedFromChannelEventDto
+import io.getstream.chat.android.client.api2.model.dto.ReactionDeletedEventDto
+import io.getstream.chat.android.client.api2.model.dto.ReactionNewEventDto
+import io.getstream.chat.android.client.api2.model.dto.ReactionUpdateEventDto
+import io.getstream.chat.android.client.api2.model.dto.TypingStartEventDto
+import io.getstream.chat.android.client.api2.model.dto.TypingStopEventDto
+import io.getstream.chat.android.client.api2.model.dto.UnknownEventDto
+import io.getstream.chat.android.client.api2.model.dto.UserDeletedEventDto
+import io.getstream.chat.android.client.api2.model.dto.UserMutedEventDto
+import io.getstream.chat.android.client.api2.model.dto.UserPresenceChangedEventDto
+import io.getstream.chat.android.client.api2.model.dto.UserStartWatchingEventDto
+import io.getstream.chat.android.client.api2.model.dto.UserStopWatchingEventDto
+import io.getstream.chat.android.client.api2.model.dto.UserUnmutedEventDto
+import io.getstream.chat.android.client.api2.model.dto.UserUpdatedEventDto
+import io.getstream.chat.android.client.api2.model.dto.UsersMutedEventDto
+import io.getstream.chat.android.client.api2.model.dto.UsersUnmutedEventDto
+import io.getstream.chat.android.client.models.EventType
+import java.lang.reflect.Type
+import java.util.Date
+
+internal class EventAdapterFactory : JsonAdapter.Factory {
+    override fun create(type: Type, annotations: MutableSet<out Annotation>, moshi: Moshi): JsonAdapter<*>? {
+        return when (type.rawType) {
+            ChatEventDto::class.java -> EventDtoAdapter(moshi)
+            else -> null
+        }
+    }
+}
+
+internal class EventDtoAdapter(
+    private val moshi: Moshi,
+) : JsonAdapter<ChatEventDto>() {
+
+    private val mapAdapter: JsonAdapter<MutableMap<String, Any?>> =
+        moshi.adapter(Types.newParameterizedType(Map::class.java, String::class.java, Any::class.java))
```

Comment:
Moshi caches adapters returned by `adapter`, but don't even want to hit the cache for all of these adapters that are used here, so saving them all into properties instead